### PR TITLE
Item & ItemMerchant Classes, Complete

### PR DESCRIPTION
**Pull request by ajongaro**
Passing tests on both classes, ready for SalesEngine (I hope!)

### CODE REVIEW

File: spec/item_repository_spec.rb [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        :merchant_id => 9}) }
+
+  let(:item_5) { Item.new({
+        :id          => 5,
+        :name        => "Mouse",
+        :description => "Moves the cursor around",
+        :unit_price  => BigDecimal(23.99,4),
+        :created_at  => Time.now,
+        :updated_at  => Time.now,
+        :merchant_id => 9}) }
```
Comment (KelsiePorter) [inline spec/item_repository_spec.rb:50]: Great idea to use let! for all of these items.

File: spec/item_repository_spec.rb [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      ir.add_to_repo(item_1)
+      ir.add_to_repo(item_2)
+      ir.add_to_repo(item_3)
+      ir.add_to_repo(item_4)
+
+      expect(ir.find_all_with_price_in_range(19..30)).to eq([item_3, item_4])
+      expect(ir.find_all_with_price_in_range(0..9)).to eq([])
+      expect(ir.find_all_with_price_in_range(19.99..21.21)).to eq([item_3])
+    end
+  end
```
Comment (KelsiePorter) [inline spec/item_repository_spec.rb:110]: Good test coverage on this one.

File: spec/item_repository_spec.rb [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+      expect(ir.all.count).to eq(3)
+
+      ir.delete(1)
+      expect(ir.all.count).to eq(2)
+      expect(ir.all).to eq([item_2, item_3])
+
+      ir.delete(3)
+      expect(ir.all).to eq([item_2])
+    end
+  end
```
Comment (KelsiePorter) [inline spec/item_repository_spec.rb:196]: Good test coverage here with that first expectation

File: lib/item_repository.rb [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    end
+  end
+
+  def find_all_by_merchant_id(merchant_id)
+    @all.find_all do |item|
+      item.merchant_id == merchant_id
+    end
+  end
+
+  # can we use find_all_with_price?
```
Comment (KelsiePorter) [inline lib/item_repository.rb]: We can probably remove this comment

File: lib/item_repository.rb [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  def create(attributes)
+    attributes[:id] = max_id
+    add_to_repo(Item.new(attributes))
+  end
+
+  def update(id, attributes)
+    item = find_by_id(id)
+    
+    attributes.each do |key, value|
+      item.instance_variable_set("@#{key}", value)
```
Comment (KelsiePorter) [inline lib/item_repository.rb:65]: does instance_variable_set end up using the attr_accessor in item.rb?

**Review by this-is-joeking — COMMENTED** [review]
Test

**Pull request merged**